src/backend/graphics/opengl.rs

```diff
@@ -0,0 +1,17 @@
+use nix::c_void;
+
+pub enum SwapBuffersError {
+    ContextLost,
+    AlreadySwapped,
+}
+
+pub trait OpenGLRenderer
+{
+    fn swap_buffers(&self) -> Result<(), SwapBuffersError>;
+    unsafe fn get_proc_address(&self, symbol: &str) -> *const c_void;
+    fn get_framebuffer_dimensions(&self) -> (u32, u32);
+    fn is_current(&self) -> bool;
+    unsafe fn make_current(&self);
+    fn get_api(&self) -> Api;
+    fn get_pixel_format(&self) -> PixelFormat;
```

Comment:
It's okay-ish. I am not happy with copying most of the stuff from glutin and glium. It works, but I think we should drop it completely. A backend should just expose it's own rendering backend.